src/utils/CharactersPageUtils.js

```diff
@@ -71,27 +71,32 @@ export const prepareNewCharacterData = ({ formData }) => {
 }
 
 export const prepareEditCharacterData = ({ formData, character }) => {
-  const {
-    name,
-    eyeColor,
-    height,
-    gender,
-    birthYear,
-    homeworld,
-    species,
-    numberOfFilms,
-  } = formData
+  const { name, eyeColor, height, gender, birthYear, species, numberOfFilms } =
+    formData
   const { id } = character
 
+  const newHomeworld =
+    formData.homeworld === character.homeworld.name
+      ? {
+          ...character.homeworld,
+        }
+      : {
+          ...character.homeworld,
+          id: nanoid(),
+          name: formData.homeworld,
+        }
+
```

Comment:
can you change the `homeworld` input to be a selector instead of a free text input